lib/TestModule.py

```diff
@@ -0,0 +1,70 @@
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cbor import cbor
+import pem
+import hexdump
+from typing import Optional, Tuple
+from aiocoap.error import NetworkError
+from aiocoap import *
+from logging import error, info
+import asyncio
+import logging
+import cbor
+import subprocess
+
+logging.basicConfig(level=logging.INFO)
+BASE_URI = 'coap://169.254.0.1/api/v1'
+
+requests = { # endpoint, method, input, output
+    '/admin/token_provision':{'method': Code.POST,'in': 'cbor','out': 'csr'},
+    '/admin/provision_complete':{'method': Code.POST,'in': 'csr','out': None}
+}
+
+class TestModule(object):
+    ROBOT_LIBRARY_VERSION = '1.0.0'
+    ROBOT_LIBRARY_SCOPE = 'GLOBAL'
+    
+    def convert_pem_to_cbor(self, input_file, output_file):
+
+        chain = pem.parse_file(input_file)
+        cert = []
+        for c in chain:
+            c = x509.load_pem_x509_certificate(
+                str(c).encode('utf-8'), default_backend())
+            der = c.public_bytes(serialization.Encoding.DER)
+            cert.append(bytes(der))
+        encoded = cbor.dumps({
+            'certs': cert
+        })
+
+        with open(output_file, 'wb') as out:
+            out.write(encoded)
+            out.close()
+
+        # hexdump.hexdump(encoded)
+
+    async def send_data(self, endpoint, argument):
+        proto = await Context.create_client_context()
```

Comment:
I add commented code in `TestModule.py` - After adding the constructor, the whole module wasn't working. After a little research, I think `async` +` __init__` won't work together. With two calls to the `send_data` function, no problem appears, but as I understand that it can appears in the future.

Reply:
I tried to use recommended library, but it didn't work correctly - I got the same problem: https://github.com/Chetic/robotframework-async/issues/5 
I used workaround from this issue: https://github.com/robotframework/robotframework/issues/4089

Reply:
@Pokisiekk I made a few improvements - see my commits